sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -604,13 +612,17 @@ namespace Azure { namespace Core { namespace Http {
    * @brief Raw HTTP response.
    */
   class RawResponse {
+  public:
+    typedef std::
+        map<std::string, std::string, Azure::Core::Internal::Strings::CaseInsensitiveComparator>
+            HeaderMap;
```

Comment:
Is it OK for a public typedef to expose an internal type? Doesn't that break our access rules. Are we allowed to rename or  change the `CaseInsensitiveComparator` if we expose it in the type definition like this?